index.js

```diff
@@ -38,8 +42,10 @@ PredictiveServiceClient.prototype.setEndpoint = function(end_point) {
   var protocol_uri = PredictiveServiceClient._getProtocol(end_point);
```

Comment:
not critical: this is including a utility function (with no dependency on the object) in an object.  Consider moving such functions out of the object and having static function declarations at the top of the file.  

``` 
// definition at top of script
function getProtocolURI(endpoint) {
...
}

... inside object
this.endpoint = getProtocolURI(endpoint);